### ISSUE TITLE
Font default validation not working correctly on subclasses

### ISSUE THREAD

**Issue by corranwebster** [enthought/enable#945]
In the following code, `b.f` is undefined when assigned at the class level, but works fine when assigned as ab attribute (so not a problem with the font):
```
In [1]: from enable.trait_defs.api import KivaFont
In [2]: from traits.api import HasTraits
In [3]: class A(HasTraits):
   ...:     f = KivaFont()
   ...: 
In [4]: class B(A):
   ...:     f = "modern 11"
   ...: 
In [5]: b = B()
In [6]: b.f
Out[6]: <undefined>
In [7]: b.f = "modern 11"
In [8]: b.f
Out[8]: Font(face_name='', size=11, family=3, weight=400, style=0, underline=False, encoding=0)
```
This might be a traits bug?

**Comment by mdickinson** [comment]
Do we have existing code that uses this pattern? The general form of this pattern hasn't worked in Traits for quite a while. As a workaround, can we recommend that people not subclass in this way, and instead do:

```
class A(HasTraits):
    f = KivaFont()
    
class B(A):
    f = KivaFont("modern 11")
```

We should of course fix this in Traits eventually, but I'm having trouble seeing this as something that needs an urgent fix in Traits when this aspect of Traits hasn't changed for quite a while.

**Comment by corranwebster** [comment]
The problem is that the breakage is in `HasTraits`/`CTrait` which we can't override (this is because we are overriding `clone` already to correctly provide a `callable_with_args` default, but `HasTraits` doesn't handle that correctly.

**Comment by corranwebster** [comment]
This is fixed by https://github.com/enthought/traits/pull/1645 which is available in Traits >= 6.4.0
